### ISSUE TITLE
add 'failed_values' column to dq_summary table

### ISSUE THREAD

**Issue by pjy953** [GoogleCloudPlatform/cloud-data-quality#172]
When using clouddq, I felt the need to check the actual failed cases in dq_summary.
So, i want to know your opinion about adding 'failed_values' column to dq_summary.

We need just a few lines to add failed_values, as follows.
**ARRAY~ ~failed_values**

main.sql
```
~
        null_count,
        null_percentage,
        ARRAY(
            SELECT column_value 
            FROM {{ ref(entity_dq_statistics_model) }} 
            WHERE column_value IS NOT NULL AND simple_rule_row_is_valid is False LIMIT 500
         ) AS failed_values
    FROM
        {{ ref(entity_dq_statistics_model) }}
~
```

**Comment by pbalm** [comment]
Are you aware that for row-based rules, you have this in the rule binding _views_? There is a column `column_value` there. Does that solve your need?

**Comment by pjy953** [comment]
Thank you for your reply. I already know about a column `column_value` in rule binding _views_. 
However, in the case of a field in the view, it is necessary to calculate each time to obtain the actual value. So I tried to store it in dq_summary as sampled data.
I thought that if I had such sample data, I would be able to check the wrong result right away without recalculation.

**Comment by thinhha** [comment]
One idea I can think of is to add a new boolean flag to each rule binding as an opt-in to materialize each rule binding as a table on each run instead of a view. This way you can query the rule binding logic result directly instead of having to query the view again.

```
rule_bindings:
  T2_DQ_1_EMAIL:
    entity_id: TEST_TABLE
    column_id: VALUE
    row_filter_id: DATA_TYPE_EMAIL
    incremental_time_filter_column_id: TS
    ***materialize_rule_binding_results: true*** # false by default
    rule_ids:
      - NOT_NULL_SIMPLE
      - REGEX_VALID_EMAIL
      - CUSTOM_SQL_LENGTH_LE_30
    metadata:
      team: team-2
```

Would this meet your need @pjy953 ?

**Comment by pjy953** [comment]
@thinhha 
Thank you for your response.
Yes. The options you suggested meets my need.
If there is no problem in adding that flag, i will upload the PR in the near future. 

Thank you.

**Comment by thinhha** [comment]
Hi @pjy953 we recently released a different solution in v1.0.1 to add a column to the results table that contains a SQL query you can execute to get the failed records:
https://github.com/GoogleCloudPlatform/cloud-data-quality/releases/tag/v1.0.1

Would you be able to check if this works for you? Please close the ticket if so.
